src/averageAges.js

```diff
@@ -36,8 +40,21 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
+function isHaveChildren(people, womenName) {
+  return people
+    .map(person => person.mother === womenName)
+    .includes(true);
+}
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = !withChildren
+    ? people.filter(({ sex }) => sex === 'f')
+    : people.filter(({ sex, name }) => sex === 'f'
+    && isHaveChildren(people, name));
```

Comment:
You have two filter calls - but a cleaner option would be a single filter call with more complex predicate